### ISSUE TITLE
Sunrise ramp-up speed different from sunset ramp-down speed?

### ISSUE THREAD

**Issue by osmaa** [basnijholt/adaptive-lighting#813]
Have you ever considered separating the adjustment speed of sunrise and sunset? This would be beneficial for using adaptive lighting for wake-up lights. For example:

Wakeup might be preferred to start at minimum brightness and color temperature, but ramp up both brightness and coldness of the light within, say, just 15 minutes synchronized to an alarm.

Wind-down on the other hand might start an hour before sunset and continue an hour after sunset to minimum brightness and warm lighting.

The tanh and linear modes could achieve this, were the morning/evening time parameters separately adjustable. OTOH, it seems that the current `brightness_mode_time_dark` and `brightness_mode_time_light` parameters are somewhat redundant given the  ramp time window can also be shifted using the sunrise and sunset offsets.

**Comment by jtcressy** [comment]
+1 on this. been struggling with a half-baked wake-up/wind-down automation for months. quite an iron-triangle situation. I want to have my cake and eat it too!

It would be nice to have something like the following options:  
* `brightness_ramp_time_sunrise`
  - Duration in seconds to transition from `min_brightness` to `max_brightness` before sunrise  
* `brightness_ramp_time_sunset`
  - Duration in seconds to transition from `max_brightness` to `min_brightness` after sunset

then as OP suggested, combine with existing options `sunrise_offset` and `sunset_offset` to shift the ramp period around the `sunrise_time` and `sunset_time`.

also, for the sake of completeness, maybe `brightness_mode` could also be split between sunrise and sunset in case you wanted to have a `linear` ramp in the morning and a `tanh` ramp at sunset.

**Comment by danishru** [comment]
+1
I'm also looking for a way to make the light temperature change at different speeds. For example, Apple Homekit adaptive lighting uses different times and speeds to adjust the light temperature depending on the position of the sun and/or time. I plotted the Apple Homekit adaptive lighting operation and compared it with these AL. AL uses sunrise and sunset exclusively, but I would also like adjustments for dawn and dusk. Are there any ways to configure AL the same way?
![image_2024-04-08_10-29-50](https://github.com/basnijholt/adaptive-lighting/assets/153666170/65b34eb2-66f2-457f-b52e-9b2521e21c91)

**Comment by basnijholt** [comment]
@danishru, could you in some way extract the numerical values for the Apple HomeKit curve? We could implement something similar for AL.

**Comment by danishru** [comment]
> @danishru, could you in some way extract the numerical values for the Apple HomeKit curve? We could implement something similar for AL.

Yes, here is the [history.csv](https://github.com/basnijholt/adaptive-lighting/files/14914348/history.csv) this curve to csv file from Home Assistant. The time in the csv is for UTC±00:00, but my chart is using UTC+3. At the beginning of the graph and in the data, the value is less than "3000" and "unknown", they can be deleted, since they appeared when the lamp was turned off.

I suspect that Apple HomeKit builds a much more complex temperature curve, and my graph is only a special case. I could not find a description of the algorithm and logic, perhaps the community can help with this. If necessary, I can repeat the values recording of the Apple HomeKit adaptive light.

If we can replicate this in AL, it will be an even better user experience.

**Comment by jmattroberts** [comment]
Came here to request this. 

I like the default mode behavior for sunset. But for sunrise, I'd prefer linear or tanh with custom time dark and time light settings. 

I would have been happy switching modes back and forth using a couple of automations, but brightness mode is not exposed there.

**Comment by AlexNPavel** [comment]
Add me to the list of people that want this feature. I like to use my lights as a sunrise alarm. The way sunset works is fine for me, but I want to be able to ramp up the color from minimum to maximum in 30 minutes, which can't be done right now.

**Comment by MrtnZ0791** [comment]
My solution is to create two instances, one for sunrise and one for sunset. Each with different settings fitting to my needs. An automation at noon switches sunrise instances off and sunset instances on and at midnight vic versa.
It's only a dirty workaround...
But i Support this feature requests, too.

**Comment by mattrobertsfr** [comment]
@MrtnZ0791 That's what I did, too, until very recently. Now I just have one configuration (1 per area instead of 2), and I use this automation to switch brightness mode using `change_switch_settings`.

[Change_Adaptive_Lighting_Mode.yaml](https://github.com/user-attachments/files/32418833/Change_Adaptive_Lighting_Mode.yaml)

**Comment by djurny** [comment]
As others, of course I also have something cooked up to address just the `brightness_mode_time_light` and `brightness_mode_time_dark`, see my automation [here](https://github.com/djurny/adaptive-lighting/tree/main/home-assistant/automations.d/Adjust%20brightness_mode_time.yaml). It will 'stretch' the `brightness_mode_time_{light|dark}` to the `max_{sunrise|sunset}_time`. Switching occurs at solar noon and solar midnight.

This has been running without an (major) issue for over a year, only gripe is that when I restart/reload HASS, the calculation for the next ramp reverts to default. I just work around that by restarting/reloading HASS before the next solar turnaround.

Having seperate 'ramp' settings for sunrise and sunset per default, along with the other settings would indeed be great.
Grt,
